### DISCUSSION TITLE
Properly configuring Tuya LED using SetOption20, device does not turn on without changed brightness

### DISCUSSION THREAD

**Discussion by pomeloy** [arendst/Tasmota#18501] [Support request]
I have a Tuya device with RGB LEDs configured as follows: 
```
{"GPIO":[0,2272,0,2304,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":54}
Backlog TuyaMCU 12,7; TuyaMCU 21,8; TuyaMCU 24,10; TuyaRGB 3
```
The device reverts to the default color and full brightness when using `Power2 1` to turn it on, so I have to use `
SetOption20 1`. This however has the side-effect of the LEDs not turning back on if the `Dimmer <x>` value has not changed in-between.

E.g. if `stat/{topic}/RESULT = {"POWER2":"OFF","Dimmer":50,"Color":[..]`, then
`Dimmer 50` would not turn it back on while `Dimmer 51` would.

That means I cannot just turn the device on and off from my smart home central, I always have to adjust the brightness, so no simply clicking the power button or using voice assistants to turn the LEDs on.

I tried coming up with a rule to circumvent this but seeing as technically the brightness value has not changed I cannot even rely on the Dimmer1#state trigger. Is there any non-hacky way to fix this?

**Comment by pomeloy** [comment]
I’m still at a loss as to how to approach this.

**Reply by sfromis** [reply]
When using a device suffering from a TuyaMCU infestation, operation is up to that other MCU, with Tasmota having no control, outside of acting like an app or remote control would be able to. That has the effect that the usual Tasmota features for managaing details cannot be expected to work seamlessly, only sometimes for basic operation.

**Reply by pomeloy** [reply]
Do you know of any workaround?

**Reply by sfromis** [reply]
Not really, but I've avoided the hassles with TuyaMCU. You'd likely be able to do something with rules, reacting to messages already being published, and taking further action as appropriate.
